cmd/runtimetest/main.go

```diff
@@ -271,6 +271,62 @@ func validateDefaultFS(spec *rspec.Spec) error {
 	return nil
 }
 
+func validateLinuxDevices(spec *rspec.Spec) error {
+	logrus.Debugf("validating linux devices")
+
+	for _, device := range spec.Linux.Devices {
+		fi, err := os.Stat(device.Path)
+		if err != nil {
+			return err
+		}
```

Comment:
On Tue, Sep 13, 2016 at 06:22:39PM -0700, Ma Shimiao wrote:
> > +	for _, device := range spec.Linux.Devices {
> > +		fi, err := os.Stat(device.Path)
> > +		if err != nil {
> > +			return err
> > +		}
> 
> On 09/14/2016 01:52 AM, W. Trevor King wrote:
> > you could remove the device before calling |start|
>
> Sorry, I'm not sure about this.  Could you tell me by how to remove
> the device?

I'm not saying that you shuold remove anything.  I'm just saying that
a valid runtime launching a valid config might not pass this test.
For example:

  {
    …
    "devices": [
      {
        "path": "/dev/fuse",
        …
      }
    ],
    "hooks": {
      "prestart": [
        {
          "path": "/bin/rm",
          "args": ["rm", "/dev/fuse"]
        }
    }
  }

That's going to lead to a container without a /dev/fuse entry, which
would fail this check.  I think that's ok because we won't be calling
runtimetest with such a config.  But I'd like to put a caveat in the
--help description to make it clear that we're not attempting to
validate all possible configs with runtimetest.